### ISSUE TITLE
devtools::install_github just hangs while installing CRAN packages works fine

### ISSUE THREAD

**Issue by schillp5** [rocker-org/rocker-versioned2#152]
I've tried both rocker/verse and rocker/tidyverse, same results: I am able to install from CRAN just fine. But anything involving install_github just hangs:

> library(devtools)
> install_github(repo = "hhoeflin/hdf5r")
Downloading GitHub repo hhoeflin/hdf5r@HEAD

**Comment by eddelbuettel** [comment]
That's possibly at your end.  I use the base containers more than the versioned one but this should work the same way:

```sh
edd@rob:~$ docker run --rm -ti rocker/r-base bash -c 'install.r remotes; installGithub.r eddelbuettel/digest'                                                                                                      trying URL 'https://cloud.r-project.org/src/contrib/remotes_2.3.0.tar.gz'
Content type 'application/x-gzip' length 148405 bytes (144 KB)                                                                                                                                                     ==================================================
downloaded 144 KB                                                                                                                                                                                                                                                      
* installing *source* package ‘remotes’ ...                                                                                                                                                                        ** package ‘remotes’ successfully unpacked and MD5 sums checked
** using staged installation                                                                                                                                                                                       ** R                   
** inst                                                                                                                                                                                                            ** byte-compile and prepare package for lazy loading 
** help                                                                                                                                                                                                            *** installing help indices
** building package indices                                                                                                                                                                                        ** installing vignettes  
** testing if installed package can be loaded from temporary location                                                                                                                                              ** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path                                                                                                                                      * DONE (remotes)             
                                                                                                                                                                                                                   The downloaded source packages are in                                                                    
        ‘/tmp/downloaded_packages’                                                                                                                                                                                 Downloading GitHub repo eddelbuettel/digest@HEAD
Running `R CMD build`...                                                                                                                                                                                           * checking for file ‘/tmp/remotes86119ec3bc/eddelbuettel-digest-5ee3cd4/DESCRIPTION’ ... OK
* preparing ‘digest’:                                                                                    
* checking DESCRIPTION meta-information ... OK
* cleaning src
* running ‘cleanup’
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘digest_0.6.27.1.tar.gz’
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
* installing *source* package ‘digest’ ...                                                               
** using staged installation                                                                             
** libs                                                                                                  
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURC
E=2 -g  -c SpookyV2.cpp -o SpookyV2.o

gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c aes.c -o aes.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c blake3.c -o blake3.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c blake3_dispatch.c -o blake3_dispatch.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c blake3_portable.c -o blake3_portable.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c crc32.c -o crc32.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c digest.c -o digest.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c digest2int.c -o digest2int.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.c -o init.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c md5.c -o md5.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c pmurhash.c -o pmurhash.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c raes.c -o raes.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sha1.c -o sha1.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sha2.c -o sha2.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sha256.c -o sha256.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c spooky_serialize.cpp -o spooky_serialize.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c xxhash.c -o xxhash.o
g++ -std=gnu++11 -shared -L/usr/lib/R/lib -Wl,-z,relro -o digest.so SpookyV2.o aes.o blake3.o blake3_dispatch.o blake3_portable.o crc32.o digest.o digest2int.o init.o md5.o pmurhash.o raes.o sha1.o sha2.o sha256.o spooky_serialize.o xxhash.o -L/usr/lib/R/lib -lR
installing to /usr/local/lib/R/site-library/00LOCK-digest/00new/digest/libs
** R
** demo
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** checking absolute paths in shared objects and dynamic libraries
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (digest)
edd@rob:~$ 
```

After installing `remotes` (on this narrower container), the installation from GitHub proceeds just fine.  Maybe you have a networking issue, maybe you need a `GITHUB_PAT`, ...  but whatever it is it is likely _not_ related to Rocker.

**Comment by schillp5** [comment]
If I do install_github from a non-docker (local) RStudio session on the same ubuntu server, I have no problems. So I can't figure out why it is not working on rocker. I wish there was some kind of verbose output I could look at when I do the install_github call.

**Comment by schillp5** [comment]
> That's possibly at your end. I use the base containers more than the versioned one but this should work the same way:
> 
> ```shell
> edd@rob:~$ docker run --rm -ti rocker/r-base bash -c 'install.r remotes; installGithub.r eddelbuettel/digest'                                                                                                      trying URL 'https://cloud.r-project.org/src/contrib/remotes_2.3.0.tar.gz'
> Content type 'application/x-gzip' length 148405 bytes (144 KB)                                                                                                                                                     ==================================================
> downloaded 144 KB                                                                                                                                                                                                                                                      
> * installing *source* package ‘remotes’ ...                                                                                                                                                                        ** package ‘remotes’ successfully unpacked and MD5 sums checked
> ** using staged installation                                                                                                                                                                                       ** R                   
> ** inst                                                                                                                                                                                                            ** byte-compile and prepare package for lazy loading 
> ** help                                                                                                                                                                                                            *** installing help indices
> ** building package indices                                                                                                                                                                                        ** installing vignettes  
> ** testing if installed package can be loaded from temporary location                                                                                                                                              ** testing if installed package can be loaded from final location
> ** testing if installed package keeps a record of temporary installation path                                                                                                                                      * DONE (remotes)         
>                                                                                                                                                                                                                    The downloaded source packages are in    
>         ‘/tmp/downloaded_packages’                                                                                                                                                                                 Downloading GitHub repo eddelbuettel/digest@HEAD
> Running `R CMD build`...                                                                                                                                                                                           * checking for file ‘/tmp/remotes86119ec3bc/eddelbuettel-digest-5ee3cd4/DESCRIPTION’ ... OK
> * preparing ‘digest’:                                                                                    
> * checking DESCRIPTION meta-information ... OK
> * cleaning src
> * running ‘cleanup’
> * checking for LF line-endings in source and make files and shell scripts
> * checking for empty or unneeded directories
> * building ‘digest_0.6.27.1.tar.gz’
> Installing package into ‘/usr/local/lib/R/site-library’
> (as ‘lib’ is unspecified)
> * installing *source* package ‘digest’ ...                                                               
> ** using staged installation                                                                             
> ** libs                                                                                                  
> g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SpookyV2.cpp -o SpookyV2.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c aes.c -o aes.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c blake3.c -o blake3.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c blake3_dispatch.c -o blake3_dispatch.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c blake3_portable.c -o blake3_portable.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c crc32.c -o crc32.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c digest.c -o digest.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c digest2int.c -o digest2int.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c init.c -o init.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c md5.c -o md5.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c pmurhash.c -o pmurhash.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c raes.c -o raes.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sha1.c -o sha1.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sha2.c -o sha2.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c sha256.c -o sha256.o
> g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c spooky_serialize.cpp -o spooky_serialize.o
> gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-oKyfjH/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c xxhash.c -o xxhash.o
> g++ -std=gnu++11 -shared -L/usr/lib/R/lib -Wl,-z,relro -o digest.so SpookyV2.o aes.o blake3.o blake3_dispatch.o blake3_portable.o crc32.o digest.o digest2int.o init.o md5.o pmurhash.o raes.o sha1.o sha2.o sha256.o spooky_serialize.o xxhash.o -L/usr/lib/R/lib -lR
> installing to /usr/local/lib/R/site-library/00LOCK-digest/00new/digest/libs
> ** R
> ** demo
> ** inst
> ** byte-compile and prepare package for lazy loading
> ** help
> *** installing help indices
> ** building package indices
> ** installing vignettes
> ** testing if installed package can be loaded from temporary location
> ** checking absolute paths in shared objects and dynamic libraries
> ** testing if installed package can be loaded from final location
> ** testing if installed package keeps a record of temporary installation path
> * DONE (digest)
> edd@rob:~$ 
> ```
> 
> After installing `remotes` (on this narrower container), the installation from GitHub proceeds just fine. Maybe you have a networking issue, maybe you need a `GITHUB_PAT`, ... but whatever it is it is likely _not_ related to Rocker.

If I do install_github from a non-docker (local) RStudio session on the same ubuntu server, I have no problems. So I can't figure out why it is not working on rocker. I wish there was some kind of verbose output I could look at when I do the install_github call.

**Comment by eddelbuettel** [comment]
Been a while since I looked into that code but AFAICR it does fairly little. You could trace it / approximate it by hitting `download.file()` "by hand" on the release tarball / zip file that GH offers.  If you get the file ... then we may have a smoking gun. Or not -- there are five or so options by now to `download.file()`. 

Just to be plain: you are not behind some corporate, academic, <whatever> firewall are you?

**Comment by schillp5** [comment]
> Been a while since I looked into that code but AFAICR it does fairly little. You could trace it / approximate it by hitting `download.file()` "by hand" on the release tarball / zip file that GH offers. If you get the file ... then we may have a smoking gun. Or not -- there are five or so options by now to `download.file()`.
> 
> Just to be plain: you are not behind some corporate, academic, firewall are you?

I am behind a corporate firewall, but like I said, I am able to devtools::install_github just fine from a local RStudio install on the same machine. So I don't think it is a networking issue...unless for some reason the rocker/verse image uses different ports than a standard install, or something?

**Comment by schillp5** [comment]
Well, in case it helps anybody, my workaround for now is: 
- get the download URL for the github master.zip file
- in the rocker/tidyverse Terminal session download it with wget, for example: wget https://github.com/hhoeflin/hdf5r/archive/refs/heads/master.zip 
- then in the R window install it with devtools::install_local("/home/rstudio/master.zip")

**Comment by eddelbuettel** [comment]
You can try setting  `options("download.file.method" = "wget")`  in R to use `wget` from it.

**Comment by cboettig** [comment]
yup, the versioned2 stack sets the default to libcurl, not wget, so changing the download.file.method should fix it. 
https://github.com/rocker-org/rocker-versioned2/blob/master/scripts/install_R.sh#L130

that said, might be worth knowing why your system blocks libcurl connections?  install_github has always worked fine for me on these images so I don't think it's an issue with libcurl not liking the GitHub SSL cert chain or something.

**Comment by eitsupi** [comment]
I'm wondering if this has something to do with the R session on RStudio not using environment variables (fixed by #186), or GitHub API limitations.

I'm closing this issue for lack of additional information, but if the problem still persists, please let me know so we can re-open it.
